sfge/src/main/java/com/salesforce/logging/LoggerUtil.java

```diff
@@ -0,0 +1,51 @@
+package com.salesforce.logging;
+
+import com.google.gson.Gson;
+import com.salesforce.graph.vertex.BaseSFVertex;
+import com.salesforce.messaging.CliMessager;
+import com.salesforce.messaging.EventKey;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+import org.apache.logging.log4j.Logger;
+
+public final class LoggerUtil {
+
+    public static void warnAndPostTelemetry(Logger logger, String msg, BaseSFVertex vertex) {
+        TelemetryData td = new TelemetryData(msg, EventType.LOG_WARNING, vertex);
+        CliMessager.postMessage("TelemetryData", EventKey.INFO_TELEMETRY, new Gson().toJson(td));
+        if (logger.isWarnEnabled()) {
+            logger.warn(msg);
+        }
+    }
+
+    /** An object that can be used as the base for an SFDX telemetry event. */
+    private static class TelemetryData {
+        /** Necessary property for telemetry objects. */
+        private final String eventName = "SFGE_TELEMETRY";
+
+        private final String message;
+        private final EventType eventType;
+        private final String vertexString;
+        private final String stackTrace;
+
+        public TelemetryData(String message, EventType eventType, BaseSFVertex vertex) {
+            this.message = message;
+            this.eventType = eventType;
+            this.vertexString = vertex.toMinimalString();
+            StackTraceElement[] stackTraceElements = Thread.currentThread().getStackTrace();
+            this.stackTrace =
+                    Arrays.stream(stackTraceElements)
```

Comment:
Get the first five elements of the stack trace, which should be enough to get a decent idea of where things are.